Name: View Source of Webpage
Version: 2.0.1-1
Author: Jason Robitaille
Description: Adds a "View Source" option to the web browser's application menu in the "Page" submenu  (MIT license)

--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -792,6 +792,7 @@
 					label: $L("Page"),
 					items: [
 						MenuData.ApplicationMenu.AddToLauncher,
+						{label:"View Source", command:"view-source-cmd"},
 						MenuData.ApplicationMenu.SharePage]
 				},
 				MenuData.ApplicationMenu.ShowBookmarks,
@@ -2120,6 +2121,18 @@
 					this.showBookmarkDialog(BookmarkDialogAssistant.createLaunchpointTask, urlReference);
 					break;
 
+				case "view-source-cmd":
+					this.controller.serviceRequest("palm://com.palm.applicationManager", {
+						method: "open",
+						parameters: {
+							id: "ca.canucksoftware.internalz",
+							params: {
+								target: "view-source://" + this.currentUrl
+							}
+						}
+					});
+					break;
+					
 				case MenuData.ApplicationMenu.SharePage.command:
 					this._shareUrl(this.currentUrl, this._getDisplayTitle(), true /*capture*/);
 					break;
